lib/models/base-locale.js

```diff
@@ -168,7 +173,11 @@ async function fetchAll() {
 }
 
 async function fetchByCommunes(codesCommunes) {
-  return mongo.db.collection('bases_locales').find({communes: {$elemMatch: {$in: codesCommunes}}}).toArray()
+  const communes = await mongo.db.collection('communes').find({code: {$in: codesCommunes}}).toArray()
```

Comment:
Tu peux directement utiliser `mongo.db.collection('communes').distinct('_bal', {code: {$in: codesCommunes}})`. Cela te renvoie le tableau des identifiants de BAL.